org.bridgedb.tools.qc/src/org/bridgedb/tools/void/VoIDTool.java

```diff
@@ -0,0 +1,153 @@
+                    new FileWriter(fileName, true));
+            out.write(str);
+            out.close();
+        }
+        catch (IOException e) {
+            System.out.println("exception occoured" + e);
+        }
+    }
+
+    public void initDatabases() throws IDMapperException{
+        String url = "jdbc:derby:jar:(" + oldDb+ ")database";
+        oldGdb = SimpleGdbFactory.createInstance("db1", url);
+        }
+        
+    public void run() throws IDMapperException, IOException {
+        appendStrToFile(fileName,"@prefix void:  <http://rdfs.org/ns/void#> .\n");
+        appendStrToFile(fileName,"@prefix pav:   <http://purl.org/pav/> .\n");
+        appendStrToFile(fileName,"@prefix freq:  <http://purl.org/cld/freq/> .\n");
+        appendStrToFile(fileName,"@prefix biopax: <http://www.biopax.org/release/biopax-level3.owl#> .\n");
+        appendStrToFile(fileName,"@prefix skos:  <http://www.w3.org/2004/02/skos/core#> .\n");
+        appendStrToFile(fileName,"@prefix rdfs:  <http://www.w3.org/2000/01/rdf-schema#> .\n");
+        appendStrToFile(fileName,"@prefix ncbigene: <http://identifiers.org/ncbigene/> .\n");
+        appendStrToFile(fileName,"@prefix pubmed: <http://www.ncbi.nlm.nih.gov/pubmed/> .\n");
+        appendStrToFile(fileName,"@prefix hmdb:  <http://identifiers.org/hmdb/> .\n");
+        appendStrToFile(fileName,"@prefix rdf:   <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .\n");
+        appendStrToFile(fileName,"@prefix gpml:  <http://vocabularies.wikipathways.org/gpml#> .\n");
+        appendStrToFile(fileName,"@prefix wp:    <http://vocabularies.wikipathways.org/wp#> .\n");
+        appendStrToFile(fileName,"@prefix dcterms: <http://purl.org/dc/terms/> .\n");
+        appendStrToFile(fileName,"@prefix dcat:  <http://www.w3.org/ns/dcat#> .\n");
+        appendStrToFile(fileName,"@prefix prov:  <http://www.w3.org/ns/prov#> .\n");
+        appendStrToFile(fileName,"@prefix foaf:  <http://xmlns.com/foaf/0.1/> .\n");
+        appendStrToFile(fileName,"\n");
+        initDatabases();
+        for (DataSource ds : oldGdb.getCapabilities().getSupportedSrcDataSources()) {
+            this.out.println(ds.getFullName());
+            createVoid(ds.getFullName());
+        }
+    }
+    public void createVoid(String db1)throws IOException{
+        db1 = db1.toLowerCase();
+        switch(db1) {
+            case "gmpl":
+                appendStrToFile(fileName,"<http://rdf.wikipathways.org/20190713/rdf/gpml> \n");
+                appendStrToFile(fileName,"        a                 dcat:Distribution ;\n");
+                appendStrToFile(fileName,"void:dataDump     <https://jenkins.bigcat.unimaas.nl/job/GPML%20to%20GPML%20+%20WP%20RDF/ws/WP2RDF/output/gpml/*zip*/gpml.zip> ;\n");
+                appendStrToFile(fileName,"dcat:downloadURL  <https://jenkins.bigcat.unimaas.nl/job/GPML%20to%20GPML%20+%20WP%20RDF/ws/WP2RDF/output/gpml/*zip*/gpml.zip> ;\n");
+                appendStrToFile(fileName,"\n");
```

Comment:
I do not think hardcoding data sources is the way forward. We want this tool to work on any arbitrary .bridge (Derby) file. Please replace the `switch` approach with code that retrieves all the information from:

1. whatever is in the Derby file
2. whatever you can learn from the `datasources.ttl`

Point 1 here links to how to store metadata in Derby files. That may need extension and I welcome your input.

Reply:
we need to store the data in the derby file while creating them, so the first thing I think I would do is change the create-bridgedb-metabolites to store the data in the derby file. Next would be to modify the VoIDTool to access the stored data in the derby file. 
I'll go through the datasources.ttl to see what I can salvage from it.